### DISCUSSION TITLE
Make it easier to kill all magit windows

### DISCUSSION THREAD

**Discussion by ghosty141** [magit/magit#4784] [Support] [answered]
Sorry for the bad title, I couldn't come up with a better one.

What I'm suggest is the following, I want to make "q" "kill all other Magit buffers associated with the current project" the default if a certain variable is set. `magit-mode-bury-buffer` with a double prefix argument does that, but I personally find using prefix arguments quite annoying thus I'd prefer a solution with a custom variable that changes the behavior of "q".

I didn't really find an easy solution to this, if this can be easily configured on a per-user basis I'd be fine with that too!

**Comment by kyleam** [comment, marked as answer]
> I didn't really find an easy solution to this, if this can be easily configured on a per-user basis I'd be fine with that too!

Perhaps not as easy as you had in mind, but you should be able to get the behavior you want with advice, something like

```elisp
(with-eval-after-load 'magit
  (advice-add
   'magit-mode-bury-buffer :around
   (lambda (fn &optional kill-buffer)
     (funcall fn
              (cond ((not kill-buffer)
                     16)
                    ((>= (prefix-numeric-value kill-buffer) 16)
                     nil)
                    (t
                     kill-buffer))))
   '((name . "magit-mode-bury-buffer--tweak-arg"))))
```

**Reply by ghosty141** [reply]
Thanks, that seems to work!
